src/test/java/teammates/test/cases/ui/BaseActionTest.java

```diff
@@ -324,7 +324,7 @@ protected void verifyAccessibleForStudentsOfTheSameCourse(String[] submissionPar
         StudentAttributes student1InCourse1 = data.students.get("student1InCourse1");
         
         gaeSimulation.loginAsStudent(student1InCourse1.googleId);
-        verifyCanAccess(submissionParams);
+        verifyCanAccess(addUserIdToParams(student1InCourse1.googleId, submissionParams));
```

Comment:
The `user` parameter shouldn't be used for any logic other than the admin's masquerade mode, otherwise anyone can pretend to be any user. If you are using the user parameter in any part of the code now, you need to remove it. Use the `account` or `student` field in `Action`